station/clients/airflow/client.py

```diff
@@ -93,11 +93,20 @@ def get_task_log(self, dag_id: str, run_id: str, task_id: str) -> dict:
         @return:
         """
 
-        last_task_try_number = next(task["try_number"] for task in self.get_run_information(dag_id, run_id)["tasklist"]["task_instances"] if task["task_id"] == task_id)
-        url = self.airflow_url + f"dags/{dag_id}/dagRuns/{run_id}/taskInstances/{task_id}/logs/{last_task_try_number}"
-        log = requests.get(url=url, auth=self.auth)
-        log.raise_for_status()
-        return log.content.decode("utf-8")
+        task_number_list = []
+        task_known = False
+        for task in self.get_run_information(dag_id, run_id)["tasklist"]["task_instances"]:
+            if task["task_id"] == task_id:
+                task_known = True
+                task_number_list.append(task["try_number"])
+        if task_known:
+            last_task_try_number = max(task_number_list)
+            url = self.airflow_url + f"dags/{dag_id}/dagRuns/{run_id}/taskInstances/{task_id}/logs/{last_task_try_number}"
+            log = requests.get(url=url, auth=self.auth)
+            log.raise_for_status()
+            return log.content.decode("utf-8")
+        else:
+            return None
```

Comment:
Optinal task try number and if not use last

Reply:
shouldn't the max of try numbers be the last execution?

Reply:
I guess, but what I was going for was an optional task_try_number parameter